dtfs-central-api/api-tests/test-api.ts

```diff
@@ -39,30 +39,30 @@ class TestApi {
 
   public post(data: object) {
     this.assertIsInitialised();
-    const to = async (url: string): Promise<Response> => await request(this.app).post(url).send(data).set(headers);
+    const to = async <T extends Response>(url: string): Promise<T> => (await request(this.app).post(url).send(data).set(headers)) as T;
     return { to };
   }
 
   public put(data: object) {
     this.assertIsInitialised();
-    const to = async (url: string): Promise<Response> => await request(this.app).put(url).send(data).set(headers);
+    const to = async <T extends Response>(url: string): Promise<T> => (await request(this.app).put(url).send(data).set(headers)) as T;
     return { to };
   }
 
-  public async get(url: string, data?: object): Promise<Response> {
+  public async get<T extends Response>(url: string, data?: object): Promise<T> {
     this.assertIsInitialised();
-    return await request(this.app).get(url).send(data).set(headers);
+    return (await request(this.app).get(url).send(data).set(headers)) as T;
   }
 
   public remove(data: object) {
     this.assertIsInitialised();
-    const to = async (url: string): Promise<Response> => await request(this.app).delete(url).send(data).set(headers);
+    const to = async <T extends Response>(url: string): Promise<T> => (await request(this.app).delete(url).send(data).set(headers)) as T;
     return { to };
   }
 
   public patch(data: object) {
     this.assertIsInitialised();
-    const to = async (url: string): Promise<Response> => await request(this.app).patch(url).send(data).set(headers);
+    const to = async <T extends Response>(url: string): Promise<T> => (await request(this.app).patch(url).send(data).set(headers)) as T;
```

Comment:
What's the reason for using `T extends Response` instead of just `Response` please?

Reply:
I think this makes our tests cleaner, but just a preference thing & happy to reset it

allows me to write 
```
        const { status, body } = await testApi.get<FacilityAmendmentResponse>(generateUrl(facilityId, anInvalidAmendmentId));
```
rather than
```
```
        const { status, body } = await testApi.get(generateUrl(facilityId, anInvalidAmendmentId)) as FacilityAmendmentResponse;
```

I looked into this in more detail & it looks like sadly supertest doesn't support using generics without using an `as` somewhere along the way (https://github.com/ladjs/supertest/issues/720)

Reply:
I think we should keep as is as we don't seem to be doing this anywhere else and want to avoid introducing new approaches/patterns.